### PR TITLE
Clean up SPDK enumeration

### LINKED ISSUE #491: Silent failure for enum test with SPDK
`xnvme_tests_enum multi --count 4 --be spdk`
This command fails silently on linux. It also fails on freebsd.

Additionally, this command doesn't fail, which is probably also not good:
`xnvme_tests_enum multi --count 4 --be vfio`

### PR DESCRIPTION
**Pull request by karlowich**
Closes #491

### CODE REVIEW

**Review by safl — APPROVED** [review]
Excellent.

File: lib/xnvme_be_spdk_dev.c [hunk 1]
```diff
@@ -551,9 +551,9 @@ enumerate_attach_cb(void *cb_ctx, const struct spdk_nvme_transport_id *trid,
 		    struct spdk_nvme_ctrlr *ctrlr, const struct spdk_nvme_ctrlr_opts *ctrlr_opts)
 {
 	struct xnvme_be_spdk_enumerate_ctx *ectx = cb_ctx;
-	const int num_ns = spdk_nvme_ctrlr_get_num_ns(ctrlr);
+	int nsid = spdk_nvme_ctrlr_get_first_active_ns(ctrlr);
 
-	for (int nsid = 1; nsid <= num_ns; ++nsid) {
+	while (nsid) {
```
Comment (safl) [inline lib/xnvme_be_spdk_dev.c:556]: How about this instead:
``for (int nsid = spdk_nvme_ctrlr_get_first_active_ns(ctrlr); nsid; nsid = spdk_nvme_ctrlr_get_next_active_ns(ctrlr, nsid)) {``

Thus, keeping the ``nsid`` local..

**Review by safl — COMMENTED** [review]
One minor thing inline, awesome stuff.

File: lib/xnvme_be_spdk_dev.c (hunk 1 above)
Comment (karlowich) [reply lib/xnvme_be_spdk_dev.c]: Done

File: lib/xnvme_be_spdk_dev.c [hunk 2]
```diff
@@ -572,10 +572,7 @@ enumerate_attach_cb(void *cb_ctx, const struct spdk_nvme_transport_id *trid,
 		}
 		ns = spdk_nvme_ctrlr_get_ns(ctrlr, nsid);
 		if (!ns) {
-			continue;
-		}
-		if (!spdk_nvme_ns_is_active(ns)) {
-			XNVME_DEBUG("FAILED: skipping inactive nsid: %d", nsid);
+			XNVME_DEBUG("FAILED: spdk_nvme_ctrlr_get_ns()");
```
Comment (safl) [inline lib/xnvme_be_spdk_dev.c:575]: How about adding the ``nsid`` to the debug output here? Might be useful?

Comment (karlowich) [reply lib/xnvme_be_spdk_dev.c]: Good catch, I agree

**Review by safl — APPROVED** [review]
lgtm

**Review by safl — APPROVED** [review]
lgtm

**Pull request merged**